### PR TITLE
Support key values from slf4j's fluent api

### PR DESCRIPTION
**Pull request by mmienko**
This should address issue https://github.com/logfellow/logstash-logback-encoder/issues/946.
The configurations are the same as the configurations for MDC.

### CODE REVIEW

File: src/main/java/net/logstash/logback/composite/loggingevent/KeyValuePairJsonProvider.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+                    String fieldName = kvpKeyFieldNames.get(kv.key);
+                    if (fieldName == null) {
+                        fieldName = kv.key;
+                    }
+                    if (!hasWrittenStart && getFieldName() != null) {
+                        generator.writeObjectFieldStart(getFieldName());
+                        hasWrittenStart = true;
+                    }
+                    StructuredArguments.keyValue(fieldName, kv.value).writeTo(generator);
```
Comment (mmienko) [inline src/main/java/net/logstash/logback/composite/loggingevent/KeyValuePairJsonProvider.java:100]: This is the crux of the change, everything else is similar to the MdcJsonProvider.

File: src/main/java/net/logstash/logback/LogstashFormatter.java [hunk 2]
```diff
@@ -222,6 +216,22 @@ public void setIncludeMdc(boolean includeMdc) {
         }
     }
 
+    public boolean isIncludeKvp() {
+        return this.kvpProvider != null;
+    }
+
+    public void setIncludeKvp(boolean includeKvp) {
+        if (isIncludeKvp() != includeKvp) {
```
Comment (osigida) [inline src/main/java/net/logstash/logback/LogstashFormatter.java:224]: let's use fail fast: 
```
if (isIncludeKvp() == includeKvp) {
  return;
}
if (includeKvp) {
   kvpProvider = new KeyValuePairJsonProvider();
     addProvider(kvpProvider);
   } else {
      getProviders().removeProvider(kvpProvider);
       kvpProvider = null;
    }
```

File: src/main/java/net/logstash/logback/composite/loggingevent/KeyValuePairJsonProvider.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        if (!this.includeKvpKeyNames.isEmpty() && !this.excludeKvpKeyNames.isEmpty()) {
+            addError("Both includeKvpKeyNames and excludeKvpKeyNames are not empty.  Only one is allowed to be not empty.");
+        }
+        super.start();
+    }
+
+    @Override
+    public void writeTo(JsonGenerator generator, ILoggingEvent event) throws IOException {
+        List<KeyValuePair> kvp = event.getKeyValuePairs();
+        if (kvp != null && !kvp.isEmpty()) {
```
Comment (osigida) [inline src/main/java/net/logstash/logback/composite/loggingevent/KeyValuePairJsonProvider.java:83]: fail fast would help here as well 
```
        if (kvp == null || kvp.isEmpty()) {
          return
        }
        ...

**Review by ferhtaydn — APPROVED** [review]
LGTM. 

@osigida can you merge if there's nothing left to address?

**Comment by philsttr** [comment]
Thank you so much for the contribution and reviews!

**Pull request merged**